x-pack/legacy/plugins/maps/public/angular/map_controller.js

```diff
@@ -364,10 +364,12 @@ app.controller(
 
       await Promise.all(getIndexPatternPromises);
       // ignore outdated results
-      if (prevIndexPatternIds !== nextIndexPatternIds) {
+      if (!_.isEqual(prevIndexPatternIds, nextIndexPatternIds)) {
```

Comment:
Why was this changed from `!==`? The check is just ensuring that the array reference did not change change during the async call. There is no need to check the array content for equality.